GirafRest.Test/Controllers/DepartmentControllerTest.cs

```diff
@@ -1,291 +1,329 @@
+            var val = result.Value as SuccessResponse<DepartmentDTO>;
+
+            //assert
+            Assert.Equal(val.Data.Name, depDTO.Name);
         }
 
         [Fact]
-        [System.Obsolete]
-        public void AddResource_ValidDepartmentNonExistingResourceInDTO_ResourceNotFound()
+        public void Department_Name_Should_Change()
         {
-            var dc = initializeTest();
-            _testContext.MockUserManager.MockLoginAsUser(_testContext.MockUsers[ADMIN_DEP_ONE]);
-            var res = dc.AddResource(DEPARTMENT_ONE, NONEXISTING).Result as ObjectResult;
-            var body = res.Value as ErrorResponse;
-
-            Assert.Equal(StatusCodes.Status404NotFound, res.StatusCode);
-            Assert.Equal(ErrorCode.ResourceNotFound, body.ErrorCode);
-        }
+            //Arranging
+            var departmentController = new MockedDepartmentController();
+            var departmentRep = departmentController._departmentRepository;
 
-        [Fact]
-        [System.Obsolete]
-        public void AddResource_ValidDepartmentValidDTONoLogin_Unauthorized()
-        {
-            var dc = initializeTest();
-            var res = dc.AddResource(DEPARTMENT_ONE, RESOURCE_ONE).Result as ObjectResult;
-            var body = res.Value as ErrorResponse;
+            DepartmentNameDTO departmentNameDTO = new DepartmentNameDTO();
+            departmentNameDTO.Name = "Mansestuen";
 
-            Assert.Equal(StatusCodes.Status403Forbidden, res.StatusCode);
-            Assert.Equal(ErrorCode.NotAuthorized, body.ErrorCode);
-        }
+            var department = new Department();
+            department.Key = 1;
+            department.Name = "DenckerHaven";
 
-        #endregion
+            //Mock
+            departmentRep.Setup(repo => repo.GetDepartmentById(department.Key)).Returns(department);
+            departmentRep.Setup(repo => repo.Update(department)).Returns(Task.CompletedTask);
 
-        #region RemoveResource
+            //act
+            var response = departmentController.ChangeDepartmentName(department.Key, departmentNameDTO);
+            var objectResult = response.Result as ObjectResult;
+            var newName = objectResult.Value as SuccessResponse <string>;
 
-        [Fact]
-        [System.Obsolete]
-        public void RemoveResource_RemoveExistingResource_Success()
-        {
-            var dc = initializeTest();
-            _testContext.MockUserManager.MockLoginAsUser(_testContext.MockUsers[ADMIN_DEP_ONE]);
-            var res = dc.RemoveResource(RESOURCE_FIVE).Result as ObjectResult;
-            var body = res.Value as SuccessResponse<DepartmentDTO>;
-
-            Assert.Equal(StatusCodes.Status200OK, res.StatusCode);
-            // Check that ressource no longer exist
-            Assert.True(!(body.Data.Resources.Any(r => r == RESOURCE_FIVE)));
+            //assert
+            Assert.True(newName.Data.Equals("Name of department changed"));

```

Comment:
Er det rigtigt forstået at det eneste vi asserter for at verificere at en ny department er lavet er at sikre os at successresponse er der?
Burde vi ikke også sikre os at departmenetet er der - hvorend det bliver storet henne

Reply:
jo, men det ville være en intergration test, fordi vi skal tilgå databasen for at gøre det på den måde. successresponse burde være nok.